torch/csrc/jit/tensorexpr/operators/misc.cpp

```diff
@@ -642,6 +642,96 @@ Tensor computeCat(
       });
 }
 
+static bool checkStackInputShape(const std::vector<BufHandle>& bufList) {
+  TORCH_INTERNAL_ASSERT(
+      bufList.size() != 0,
+      buildErrorMessage("Empty input list is passed to aten::stack"));
+
+  auto buf0 = bufList.at(0);
+  TORCH_INTERNAL_ASSERT(
+      buf0.node()->dims().size() > 0, buildErrorMessage("Invalid buf rank"));
+  // Check if any of the dims is 0
+  bool hasEmptyDims = false;
+  std::vector<ExprHandle> dims;
+  for (const auto& dim : buf0.dims()) {
+    if (dim.AsNode<LongImm>() && immediateAs<int64_t>(dim) == 0ll) {
+      hasEmptyDims = true;
+    }
+    dims.push_back(IRSimplifier::simplify(dim));
+  }
+
+  // Check if all input bufs have the same size
+  for (int i = 1; i < bufList.size(); i++) {
+    auto buf = bufList.at(i);
+    TORCH_INTERNAL_ASSERT(
+        buf.node()->dims().size() == dims.size(),
+        buildErrorMessage("aten::stack inputs are with different sizes"));
+    auto dims_to_cmp = buf.dims();
+    for (int k = 0; k < dims.size(); k++) {
+      auto diff = IRSimplifier::simplify(
+          alloc<Sub>(dims_to_cmp.at(k).node(), dims.at(k).node()));
+      TORCH_INTERNAL_ASSERT(
+          diff->isConstant() && immediateAs<int>(diff) == 0,
+          buildErrorMessage("aten::stack inputs are with different sizes"));
+    }
+  }
+
+  return hasEmptyDims;
+}
+
+Tensor computeStack(
+    const std::vector<ArgValue>& inputs,
+    const std::vector<ExprHandle>& outputShape,
+    const c10::optional<ScalarType>& outputType,
+    at::Device device) {
+  // NOLINTNEXTLINE(performance-unnecessary-copy-initialization)
+  auto inputList = c10::get<BufList>(inputs[0]);
+  auto argDim = inputs[1];
+  auto hasEmptyInputs = checkStackInputShape(inputList);
+  ScalarType highType = inputList[0].dtype().scalar_type();
+  for (const auto& input : inputList) {
+    auto maybe_dtype = input.dtype().scalar_type();
+    highType = promoteTypes(highType, maybe_dtype);
+  }
+
+  return Compute(
+      "aten_stack", outputShape, [&](const std::vector<VarHandle>& axes) {
+        if (hasEmptyInputs) {
+          return ExprHandle(0);
+        }
+
+        int64_t dim_ = c10::get<int64_t>(argDim);
+        auto dim = normalizeAndCheckIndex(dim_, axes.size());
```

Comment:
It looks like function errors out when `dim_ == axes.size()`. Isn't that a correct input for `stack`?

Reply:
The `axes` are for the output of `aten::stack`. For example, 
```
a = [1, 2]
b = [3, 4]
output = aten::stack([a, b], dim=dim_)
```
`axes.size()` equals to 3, and `dim_` should be a number between 0 to 2, or -1 to -3, right? It cannot be 3. We have such dim tests for `stack`.

Reply:
Aah, I missed the point that `axes` is for the output of `stack`. Makes sense.